### DISCUSSION TITLE
Is it possible to run Vaultwarden HA?

### DISCUSSION THREAD

**Discussion by SriRamanujam** [dani-garcia/vaultwarden#1889] [Q&A]
I'd like to run multiple replicas of the Vaultwarden server for availability purposes. My server is configured to use an external Postgres database so there is no issue of having to set up storage, but I cannot find any information regarding whether multiple servers will coordinate talking to the same backend database without issues. Has anybody else seen an HA setup work for them?

**Comment by jjlin** [comment]
See e.g., https://vaultwarden.discourse.group/t/is-the-bitwarden-rs-service-stateless-if-its-using-postgres-backend/450

It should work fine, except you need a shared filesystem for attachments/sends if you're going to use those, and WebSocket notifications won't fully work since there is no communication between replicas. Replicas should also share `rsa_key.*` files between them.
